lib/utils/replace.go

```diff
@@ -197,6 +197,9 @@ func KubeResourceMatchesRegex(input types.KubernetesResource, resources []types.
 			// This means that if the user has access to pods in the "foo" namespace,
 			// they should be able to see the "foo" namespace in the list of namespaces
 			// but only if the request is read-only.
+			if isDeny && resource.Kind != types.Wildcard {
```

Comment:
Nitpick: the definition of `isDeny` is quite  a few lines of code above, and it's only used here, which decreases readability. Consider moving it closer to this line, or at least right before the for loop starts.